app/javascript/controllers/tom_select_controller.js

```diff
@@ -0,0 +1,48 @@
+import { Controller } from '@hotwired/stimulus'
+import TomSelect from 'tom-select'
+
+export default class extends Controller {
+    static values = {
+        options: {},
+        variant: String,
+        itemName: String
+    }
+
+    connect() {
+        const options = createOptions(this.optionsValue, this.variantValue, this.itemNameValue)
+
+        this.tomselect = new TomSelect(
+            this.element,
+            options
+        );
+    }
+}
+
+const createOptions = (options, variant = null, itemName = null) => {
+    const default_options = {
+        render:{
+            option_create: ( data, escape ) => {
+                return `<div class="create">Lag ny${ itemName ? ' ' + itemName : ''}: <strong>${escape(data.input)}</strong>&hellip;</div>`;
+            },
+            no_results: () => {
+                return '<div class="no-results">Ingen match</div>';
+            },
+        }
+    }
+
+    const variant_options = {
+        "dropdown": {
```

Comment:
Makes it possible to create variants like "dropdown" that have the same configurations for all variants.